src/sections/layer_and_mask_information_section/layer.rs

```diff
@@ -35,15 +82,17 @@ pub struct PsdLayer {
     pub(crate) psd_width: u32,
     /// The height of the PSD
     pub(crate) psd_height: u32,
+    /// If layer represents a group, contains sub layers.
```

Comment:
So, group is layer too, it have it own properties(opacity, for example).

Maybe it should be something like
```rust
enum PsdLayer {
     Image(ImageLayer),
     Group(GroupLayer),
     // TODO: other types
}
```

Reply:
I feel like conceptually a group is distinct from a layer (even though the PSD spec treats it as a layer).

It seems like the data is a stack of layers, some of which are in different groups.

And a group can have layer properties properties apply to the layers inside of it, but a group isn't a layer and there are some subtle differences.

For example - it looks like a group has a default blend mode of `Pass Through`, whereas `Pass Through` isn't applicable to

Maybe something like this?

```rust
// pseudocode

struct PsdLayer {
  name: String,
  layer_properties: LayerProperties,
  // ...
}

struct PsdGroup {
  name: String,
  layer_properties: LayerProperties,
  contained_layers: Range<usize>
}
```

Keeping these distinct also feels like the right choice for making it easier in the long term for us to power editing of PSD files, not just parsing.

An example of this would be someone being able to use `psd` to power a PSD editor in the browser. Sort of like a more advanced version of the [live demo](https://chinedufn.github.io/psd/drag-drop-demo/).

Someone using the API would expect groups and layers to be distinct concepts, I would think.

---

Anyways, those are just my thoughts - curious about your opinions?